### PR TITLE
fixed sidenav issue, nodata found ui issue and search issue with upda…

### PR DESCRIPTION
**Pull request by Arti-Khillare1**
…ted card component ui.

### CODE REVIEW

File: projects/lib-shared-modules/src/lib/components/card/card.component.html [hunk 1]
```diff
@@ -26,14 +31,31 @@
...
                     <div class="flex flex-column">
                         <p class="text capitalize overflow-ellipsis" [matTooltip]="list.title | translate">{{list.title | translate}}</p>
                         <div class="flex items-center text-sm text-[$borderGray]">
                             By <span class="ml-1 overflow-ellipsis">{{ list.creator | translate }}</span>
                         </div>
                     </div>
-
+                    @if(list.type === 'Program'){
+                        <div class="flex flex-row text-sm text-[$borderGray]"> 
+                            <div class="flex font-semibold">Start date:</div><span class="overflow-ellipsis">24/07/24</span>
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/card/card.component.html]: why dates are hard code ?

File: projects/lib-shared-modules/src/lib/components/filter/filter.component.html [hunk 2]
```diff
@@ -11,7 +11,7 @@
     }
     @else {
         <div>
-          <button class="inline-flex items-center text-primary px-2">
+          <button class="inline-flex items-center text-primary px-2" (click)="onButtonClick(filter)">
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/filter/filter.component.html]: please give meaning full names

File: projects/lib-shared-modules/src/lib/components/filter/filter.component.ts [hunk 3]
```diff
@@ -18,12 +18,12 @@ interface FilterChangeEvent {
 })
 export class FilterComponent {
   @Input() filterData:any;
-  @Input() showChangesButton: boolean = false;
   @Output() filteredData = new EventEmitter<FilterChangeEvent | { sort_by: string, sort_order: string }>();
   @Output() sortOptionsChanged = new EventEmitter<{ sort_by: string, sort_order: string }>();
   changeButton: string = "CHANGES_REQUIRED";
   @Input() changeReqCount: number = 0
   @Input() inprogressCount : number = 0
+  @Output() filterButtonClickEvent = new EventEmitter<{ label: string }>();
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/filter/filter.component.ts]: please give meaning full names

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="w-full p-6 flex flex-col gap-2">
     <lib-search (searchEvent)="receiveSearchResults($event)" [maxLength]="150"></lib-search>
-    <lib-filter [filterData]="filters.filterData" (filteredData)="onFilterChange($event)" (sortOptionsChanged)="onSortOptionsChanged($event)" [showChangesButton]="filters.showChangesButton" [changeReqCount]="filters.changeReqCount" [inprogressCount]="filters.inprogressCount"></lib-filter>
+    <lib-filter [filterData]="filters.filterData" (filteredData)="onFilterChange($event)" (sortOptionsChanged)="onSortOptionsChanged($event)"  [changeReqCount]="filters.changeReqCount" [inprogressCount]="filters.inprogressCount" (filterButtonClickEvent)="filterButtonClick($event)"></lib-filter>
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html]: please give more meaningful names

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html [hunk 5]
```diff
@@ -12,14 +12,16 @@
    </div>
     @for (list of filters.filteredLists; track list) {
         <div class="pb-1">
-        <lib-card [list]="list" [showActionButton]="filters.showActionButton" (buttonClickEvent)="handleButtonClick($event)"></lib-card>
+        <lib-card [list]="list" [showActionButton]="filters.showActionButton" (buttonClickEvent)="cardButtonClick($event)"></lib-card>
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html]: please give meaningful names

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html [hunk 6]
```diff
@@ -12,14 +12,16 @@
...
         </div>
     } 
     } 
     @else 
     {
         <div class="flex flex-col  justify-center items-center text-center">
-            <lib-no-result-found imagePath="assets/images/empty-state.png" [noResultMessage]="noResultMessage"></lib-no-result-found>
+            @if(filters.filteredLists.length === 0 && isDataLoaded){
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html:22]: why we need isDataLoaded ?

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts [hunk 7]
```diff
@@ -207,6 +211,17 @@ export class ResourceHolderComponent implements OnInit{
     }
   }
 
+  filterButtonClick(event : any) {
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts]: names shouldn't be "click" please rename it to "ClickEvent"

File: tailwind.config.js [hunk 8]
```diff
@@ -12,7 +12,10 @@ module.exports = {
         'chipPurple':'#A59EFB',
         'chipRejected':'#FD6969',
         'chipPublished':'#008840',
-        'numberCircle': '#EC555D'
+        'numberCircle': '#EC555D',
+        'chipInprogress': '#FEEC8C',
+        'blackText': '#000000',
```
Comment (ramkumar-ns-dev) [inline tailwind.config.js]: why black color given in color code.

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html (hunk 6 above)
Comment (Arti-Khillare1) [reply projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.html:22]: to check the api response received or not so to show nodata found ui once api response occurs only.

File: projects/lib-shared-modules/src/lib/components/search/search.component.html [hunk 9]
```diff
@@ -1,6 +1,11 @@
 <div class="search-container flex flex-row p-1 gap-2" >
-    <div class="flex text-center">
-        <mat-icon class="search-icon" matPrefix (click)="onSearch(searchInput.value)">search</mat-icon>
+    <div class="flex text-center flex-grow">
+        <mat-icon class="search-icon my-1 ml-2 cursor-pointer" matPrefix (click)="onSearch(searchInput.value)">search</mat-icon>
     </div>
-    <input matInput #searchInput [formControl]="searchControl" [attr.maxlength]="maxLength"  (keyup.enter)="onSearch(searchInput.value)" placeholder="Search" >
+    <input type="text" matInput #searchInput [formControl]="searchControl" [attr.maxlength]="maxLength"  (keyup.enter)="onSearch(searchInput.value)" placeholder="Search" class="w-full pl-1 pr-8 py-1" >
```
Comment (ramkumar-ns-dev) [inline projects/lib-shared-modules/src/lib/components/search/search.component.html:5]: https://www.w3schools.com/tags/att_input_type_search.asp can you please check this and try implement this. please check the impact if any

File: projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts [hunk 10]
```diff
@@ -207,6 +221,17 @@ export class ResourceHolderComponent implements OnInit{
     }
   }
 
+  filterButtonClickEvent(event : any) {
+    switch(event.label) {
+      case 'CHANGES_REQUIRED':
+        console.log('CHANGES_REQUIRED', event);
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts]: please remove console.logs

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/lib-shared-modules/src/lib/components/search/search.component.html (hunk 9 above)
Comment (Arti-Khillare1) [reply projects/lib-shared-modules/src/lib/components/search/search.component.html:5]: this is not working properly as the cross icon on search bar is  hides while clicking outside the searchbar.

**Pull request merged**